### PR TITLE
Mats happy-thoughts project 

### PR DESCRIPTION
**Pull request by mats-johansen**
# Happy Thoughts

Creating a kind of Twitter app with happy thoughts using React by fetching and posting to an API.

## The problem

Problem with styling being different in safari and chrome and some issues to get the like button to work properly. 

## View it live
https://wonderful-swanson-7acffc.netlify.app/

### CODE REVIEW

File: code/src/PostThoughts.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        >
+        </textarea>
+        <p className={((message.length < 5 || message.length > 140) ? "text-invalid" : "text-valid")} tabIndex='0'>
+          {message.length} / 140
+        </p>
+        <button
+          className='submit-button'
+          type='submit'
+          onClick={handleSubmit}
+          disabled={message.length < 5 || message.length > 140 ? true : false}
```
Comment (karinnordkvist) [inline code/src/PostThoughts.js:49]: This is merely a personal reflection, but I've come across a lot of discussions on using disabled buttons from an accessibility point of perspective. Here's a short text on it, if you're interested: 
[Is it ok to ‘grey out’ disabled buttons?](https://uxdesign.cc/is-it-ok-to-grey-out-disabled-buttons-8afa74a0fae)

File: code/src/PostThoughts.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        >
+        <span>
+          Send stoke here!
+        </span>
```
Comment (karinnordkvist) [inline code/src/PostThoughts.js:53]: No need for a span here, I think 😄

File: code/src/index.css [hunk 3]
```diff
@@ -1,13 +1,25 @@
...
+  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New", monospace;
 }
+
+* {
+  margin: 0;
+  padding: 0;
+}
+
+*:focus {
+  outline: 1px solid black;
```
Comment (karinnordkvist) [inline code/src/index.css:24]: 👍

File: code/src/App.js [hunk 4]
```diff
@@ -1,9 +1,12 @@
-import React from 'react'
+import React from 'react';
+import { HappyThoughts } from './HappyThoughts';
```
Comment (karinnordkvist) [inline code/src/App.js:2]: Great all over consistency in './'-paths! 👍

**Review by karinnordkvist — CHANGES_REQUESTED** [review]
Mats! Really good work on this week's project. It's clear that you were aiming for black level. 😊 I really like that you've found a smaller alternative to Moment.js, and that you used the local storage to store the likes. Great job! 👍 

You've done a really nice job with all functionality and overall your code easy to read. 😊However, I've left a few comments on the clarity of your class naming. A couple of times I had to use the devtools to find where things were located on the site. Naming your classes a little clearer (maybe with BEM-principles?) will make the code much easier to read. 😃 

Also- when trying to push the like button, I get error 400, something must have happened there. 

Keep up the great work and your level of ambition! 🌟

File: code/src/HappyThoughts.css [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  height: 120px;
+  margin: 20px;
+  padding: 20px;
+  background: rgb(245, 241, 241);
+  border: 1px solid black;
+  border-radius: 2px;
+  box-shadow: 2px 2px 0px 0px black;
+}
+
+.happy-thoughts h2 {
```
[the remark this answers is not in the document]
Comment (EmOlsz) [reply code/src/HappyThoughts.css:15]: 👍

File: code/src/LikeButton.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+export const LikeButton = props => {
+  const [clicks, setClicks] = useState((props.hearts));
+
+  if(!localStorage[props.id]) {
+    localStorage.setItem(props.id, 0)
+  }
+
+  const handleClick = () => {
+
+    fetch(`https://happy-thoughts-technigo.herokuapp.com/thoughts`, {
```
Comment (EmOlsz) [inline code/src/LikeButton.js:14]: This is not proper URL for like button, you never send POST request to API in regards to increase amount of likes by one. Please, take a look on project brief and find proper endpoint (URL) for that

File: code/src/PostThoughts.js (hunk 2 above)
Comment (EmOlsz) [reply code/src/PostThoughts.js:53]: Yep!

File: code/src/LikeButton.js (hunk 6 above)
Comment (mats-johansen) [reply code/src/LikeButton.js:14]: Fixed.

File: code/src/PostThoughts.css [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    margin: 0 0 5px 0;
+    text-transform: uppercase;
+  }
+  
+  .form-input {
+    font-family: 'Courier New', Courier, monospace;
+    font-size: 11px;
+    border: 1px solid grey;
+    padding: 5px;
+    max-width: 300px;
```
[the remark this answers is not in the document]
Comment (mats-johansen) [reply code/src/PostThoughts.css:27]: Oh nice solution! thank you

File: code/src/CreatePost.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+
+export const CreatePost = props => {
+  const { message, hearts, createdAt } = props.thought
+
+  return (
+    <div className="happy-thoughts">
+      <h2
+        className='thought-message'
+        tabIndex='0'
```
[the remark this answers is not in the document]
Comment (mats-johansen) [reply code/src/CreatePost.js:14]: Removed it!

File: code/src/CreatePost.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+      >
+        {message}
+      </h2>
+      <div className="bottom-content">
+        <LikeButton
+          hearts={hearts}
+          id={props.thought._id}
+          onLiked={props.onLiked}
```
[the remark this answers is not in the document]
Comment (mats-johansen) [reply code/src/CreatePost.js:22]: onLiked={props.onLiked} is connected to the LikeButton and without the id={props.thought._id} I get 400 (Bad Request) after I changed my URL on the LikeButton to the correct one as Maks pointed out. I'm leaving it like this even if I dont quite get it for now. Will read up on this later on

File: code/src/HappyThoughts.css (hunk 5 above)
[the remark this answers is not in the document]
Comment (mats-johansen) [reply code/src/HappyThoughts.css:15]: Fixed

File: code/src/HappyThoughts.css [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+
+.thought-info {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+}
+
+@media (min-width: 669px) {
+  .happy-thoughts {
+    width: 510px;
```
[the remark this answers is not in the document]
Comment (mats-johansen) [reply code/src/HappyThoughts.css:30]: Yeah but I am so not consistent with this between projects :D

File: code/src/LikeButton.js [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+  });
+};
+
+return (
+  <div className='like-container'>
+    <span>
+      <button
+        className='like-button'
+        type='submit'
+        onClick={handleClick}
```
[the remark this answers is not in the document]
Comment (mats-johansen) [reply code/src/LikeButton.js:31]: Fixed

File: code/src/PostThoughts.js [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+
+    fetch(THOUGHTS_URL,
+      {
+        method: 'POST',
+        headers: { 'content-type': 'application/json' },
+        body: JSON.stringify({ message })
+      }
+    ).then(() => {
+      setMessage('')
+      window.location.reload()
```
[the remark this answers is not in the document]
Comment (mats-johansen) [reply code/src/PostThoughts.js:19]: Good point !

File: code/src/PostThoughts.js (hunk 1 above)
Comment (mats-johansen) [reply code/src/PostThoughts.js:49]: Thank you, will check this out

File: code/src/PostThoughts.js (hunk 2 above)
Comment (mats-johansen) [reply code/src/PostThoughts.js:53]: Fixed

**Pull request closed without merging**